src/utils/metrics/taskMetrics.ts

```diff
@@ -64,7 +64,7 @@ export class TaskMetrics {
     }
   }
 
-  public trackTaskProcessing(jobType: string, taskType: string): { success: () => void; failure: (errorType: string) => void } | void {
+  public trackTaskProcessing(jobType: string, taskType: string): TaskProcessingTracker | undefined {
```

Comment:
`TaskProcessingTracker` already might be undefined so we can get rid of | undefined